solutions/instances/main.tf

```diff
@@ -3,10 +3,11 @@
 #######################################################################################################################
 
 locals {
-  archive_api_key = var.log_archive_api_key == null ? var.ibmcloud_api_key : var.log_archive_api_key
+  archive_api_key    = var.log_archive_api_key == null ? var.ibmcloud_api_key : var.log_archive_api_key
+  default_cos_region = var.cos_region != null ? var.cos_region : var.region
```

Comment:
what about the case where `existing_cos_instance_crn` is passed - you will need to parse the region from this

Reply:
nevermind, an instance is global so wont have a region